File: be/src/tasks/dto/create-task.dto.ts
import { IsString, IsOptional, IsDateString, IsEnum, IsUUID, IsArray } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';

export class CreateTaskDto {
  @ApiProperty({ example: 'Tên task' })
  @IsString()
  title: string;

  @ApiProperty({ example: 'Mô tả task' })
  @IsOptional()
  @IsString()
  description?: string;

  @ApiProperty({ example: '2025-01-01' })
  @IsOptional()
  @IsDateString()
  startDate?: string;

  @ApiProperty({ example: '2025-01-01' })
  @IsOptional()
  @IsDateString()
  dueDate?: string;

  @ApiProperty({ example: 'todo' })
  @IsOptional()
  @IsEnum(['todo', 'doing', 'done'])
  status?: 'todo' | 'doing' | 'done';

  @ApiProperty({ example: 'low' })
  @IsOptional()
  @IsEnum(['low', 'medium', 'high'])
  priority?: 'low' | 'medium' | 'high';

  @ApiProperty({ example: '123e4567-e89b-12d3-a456-426614174000' })
  @IsUUID()
  columnId: string;

  @ApiProperty({ example: ['123e4567-e89b-12d3-a456-426614174000'] })
  @IsOptional()
  @IsArray()
  assigneeIds?: string[];

  @ApiProperty({ example: ['123e4567-e89b-12d3-a456-426614174000'] })
  @IsOptional()
  @IsArray()
  labelIds?: string[];
}

File: backend-java/src/main/java/com/qlda/backend/tasks/dto/CreateTaskDto.java
package com.qlda.backend.tasks.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class CreateTaskDto {
    @NotBlank(message = "Title không được để trống")
    private String title;

    private String description;
    private LocalDateTime startDate;
    private LocalDateTime dueDate;

    @Pattern(regexp = "todo|done", message = "Status phải là todo hoặc done")
    private String status = "todo";

    @Pattern(regexp = "low|medium|high", message = "Priority phải là low, medium hoặc high")
    private String priority = "medium";

    @NotBlank(message = "Column ID không được để trống")
    private String columnId;

    private List<String> assigneeIds;
    private List<String> labelIds;
}

